packages/aws-cdk/lib/version.ts

```diff
@@ -0,0 +1,105 @@
+import { exec as _exec } from 'child_process';
+import colors = require('colors/safe');
+import { close as _close, open as _open, stat as _stat } from 'fs';
+import semver = require('semver');
+import { promisify } from 'util';
+import { debug, print, warning } from '../lib/logging';
+import { formatAsBanner } from '../lib/util/console-formatters';
+
+const ONE_DAY_IN_SECONDS = 1 * 24 * 60 * 60;
+
+const close = promisify(_close);
+const exec = promisify(_exec);
+const open = promisify(_open);
+const stat = promisify(_stat);
+
+export const DISPLAY_VERSION = `${versionNumber()} (build ${commit()})`;
+
+function versionNumber(): string {
+  return require('../package.json').version.replace(/\+[0-9a-f]+$/, '');
+}
+
+function commit(): string {
+  return require('../build-info.json').commit;
+}
+
+export class TimestampFile {
+  private readonly file: string;
+
+  // File modify times are accurate only till the second, hence using seconds as precision
+  private readonly ttlSecs: number;
+
+  constructor(file: string, ttlSecs: number) {
+    this.file = file;
+    this.ttlSecs = ttlSecs;
+  }
+
+  public async hasExpired(): Promise<boolean> {
+    try {
+      const lastCheckTime = (await stat(this.file)).mtimeMs;
+      const today = new Date().getTime();
+
+      if ((today - lastCheckTime) / 1000 > this.ttlSecs) { // convert ms to secs
+        return true;
+      }
+      return false;
+    } catch (err) {
+      if (err.code === 'ENOENT') {
+        return true;
+      } else {
+        throw err;
+      }
+    }
+  }
+
+  public async update(): Promise<void> {
+    const fd = await open(this.file, 'w');
```

Comment:
One more thing -- since you're writing this in the CDK tool direction, this might fail if the CDK is being run from a readonly filesystem.

Catch any exceptions please.

Reply:
Yep. There are several other edge cases that can occur due to variations on how this package is installed or the environment is configured, including `npm` not available in the current shell's `PATH`, repository not reachable, etc. 

I have a global catch from line 102-104 for this purpose.

Reply:
Makes sense.